packages/unplugin-vuetify/src/index.ts

```diff
@@ -0,0 +1,202 @@
+import type { PluginOption } from 'vite'
+import process from 'node:process'
+import fs from 'node:fs'
+import fsp from 'node:fs/promises'
+import { pathToFileURL } from 'node:url'
+import { resolveVuetifyBase } from './utils'
+import { isAbsolute, relative as relativePath } from 'pathe'
+import path from 'upath'
+import { version as VITE_VERSION } from 'vite'
+
+export interface VuetifyStylesOptions {
+  /**
+   * What CSS SASS/SCSS API should the plugin register?
+   * - `false` - don't register any SASS/SCSS API
+   * - `modern` - register SASS/SCSS 'modern' API => when using `sass`
+   * - `modern-compiler` - register SASS/SCSS 'modern-compiler' API => when using `sass-embedded`
+   *
+   * When using `modern` API, the plugin will enable `preprocessorMaxWorkers` in Vite CSS config.
+   *
+   * @default 'modern-compiler'
+   */
+  registerApi?: 'modern' | 'modern-compiler' | false
+  /**
+   * Mode to use for styles:
+   * - `none`: remove all style imports
+   * - `source`: import sass/scss from source (old `sass` option)
+   * - `configFile`: customising variables
+   */
+  mode?: true | 'none' | 'source' | {
+    configFile: string,
+  }
+}
+
+export function VuetifyStylesVitePlugin(options: VuetifyStylesOptions = {}) {
+  let configFile: string | undefined
+  // let cacheDir: string | undefined
+  const vuetifyBase = resolveVuetifyBase()
+  const noneFiles = new Set<string>()
+  let isNone = false
+  let sassVariables = false
+  let fileImport = false
+  const PREFIX = 'vuetify-styles/'
+  const SSR_PREFIX = `/@${PREFIX}`
+  const resolveCss = resolveCssFactory()
+  const api = options.registerApi ?? 'modern-compiler'
+
+  const [major, minor, patch] = VITE_VERSION.split('.')
+      .map((v: string) => v.includes('-') ? v.split('-')[0] : v)
+      .map(v => Number.parseInt(v))
+
+  return <PluginOption>{
+    name: 'unplugin-vuetify:styles',
+    enforce: 'pre',
+    config(config) {
+      if (!api)
+        return null
+
+      if (api === 'modern-compiler') {
+        return {
+          css: {
+            preprocessorOptions: {
+              sass: { api },
+              scss: { api },
+            },
+          },
+        }
+      }
+
+      if (config.css && !('preprocessorMaxWorkers' in config.css)) {
+        return {
+          css: {
+            preprocessorOptions: {
+              sass: { api },
+              scss: { api },
+            },
+          },
+        }
+      }
+
+      return {
+        css: {
+          preprocessorOptions: {
+            sass: { api },
+            scss: { api },
+          },
+          preprocessorMaxWorkers: true,
+        },
+      }
+    },
+    configResolved(config) {
+      if (config.plugins.findIndex(plugin => plugin.name === 'vuetify:styles') > -1)
+        throw new Error('The "vite-plugin-vuetify" styles plugin is incompatible with this plugin. Please remove "vite-plugin-vuetify" or set the styles to \'none\' in your Vite configuration file.')
+
```

Comment:
I guess we can remove this protection, will be transformed by styles plugin or the new one...

Reply:
dependeing on the plugin order will apply first one and may have some race conditions, message updated